Dockerfile

```
@@ -56,3 +66,86 @@ ENTRYPOINT ["docker-entrypoint.sh"]
 # # By default you'll get a single-node development server that stores everything
 # # in RAM and bootstraps itself. Don't use this configuration for production.
 CMD ["server", "-dev"]
+
+
+## UBI DOCKERFILE ##
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 as ubi
+
+ARG BIN_NAME
+# NAME and PRODUCT_VERSION are the name of the software in releases.hashicorp.com
+# and the version to download. Example: NAME=vault PRODUCT_VERSION=1.2.3.
+ARG NAME=vault
+ARG PRODUCT_VERSION
+# TARGETARCH and TARGETOS are set automatically when --platform is provided.
+ARG TARGETOS TARGETARCH
+
+# Additional metadata labels used by container registries, platforms
+# and certification scanners.
+LABEL name="Vault" \
+      maintainer="Vault Team <vault@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=${PRODUCT_VERSION} \
+      release=${PRODUCT_VERSION} \
```

I'm not sure what "release" means here, but could it be helpful to have `revision=${PRODUCT_REVISION}` here in the metadata?
```suggestion
      release=${PRODUCT_VERSION} \
      revision=${PRODUCT_REVISION} \
```

---

Release is `PRODUCT_VERSION`, set via CRT: https://github.com/hashicorp/vault/pull/15272/files/8ba72bd42342ad2d01e65d1d11641a08cff93964#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R280  Because we set version and not revision, I think that means PRODUCT_REVISION isn't useful for us to set?

---

As far as I know more labels are fine from RedHat's perspective. Their certification preflight tool does require these labels to be present: name, vendor, version, release, summary, description. [[RedHat docs](https://access.redhat.com/documentation/en-us/red_hat_software_certification/8.45/html/red_hat_openshift_software_certification_policy_guide/assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction#con-image-metadata-requirements_openshift-sw-cert-policy-container-images)]

---

Ah, should release be set to `PRODUCT_REVISION` then, or some combination of the two?

---

Oh that's an interesting idea, `PRODUCT_REVISON` is the git sha, right? So that would seem to fit in the description for `release`.

---

Got it, so `release` is a Red Hat required field.. Seems worth using `PRODUCT_REVISION` there I guess, at least it's some extra info rather than just repeating the version?

---

```suggestion
      release=${PRODUCT_REVISION} \
      revision=${PRODUCT_REVISION} \
```